src/rsz/src/SizeDownMove.cc

```diff
@@ -172,168 +171,271 @@ bool SizeDownMove::doMove(const Path* drvr_path,
-                     network_->pathName(fanout_pin),
-                     swappable->name(),
-                     new_drvr_load_cap,
-                     max_cap);
-          continue;
-        }
-      }
+      // Track worst delay change across all outputs
+      worst_delay_change = std::max(worst_delay_change, delay_change);
+    }
 
-      float new_drvr_delay
-          = resizer_->gateDelay(drvr_port, new_drvr_load_cap, dcalc_ap);
-      float new_fanout_delay = resizer_->gateDelay(
-          new_fanout_output_port, fanout_load_cap, dcalc_ap);
-      debugPrint(
-          logger_,
-          RSZ,
-          "size_down",
-          3,
-          " new delay {} FO={} drvr_delay {}<{}, fanout_delay: {}<{} ({}+{}) ",
-          network_->pathName(fanout_pin),
-          swappable->name(),
-          delayAsString(new_drvr_delay, sta_, 3),
-          delayAsString(drvr_delay, sta_, 3),
-          delayAsString(new_fanout_delay, sta_, 3),
-          delayAsString(fanout_delay - fanout_slack, sta_, 3),
-          delayAsString(fanout_delay, sta_, 3),
-          delayAsString(fanout_slack, sta_, 3));
-      // Check if the combined new delay is better than the old one
-      // If fanout slack is positive, this allows us to slow down the fanout
-      // gate If fnaout slack is negative, this allows us to accept it as long
-      // as the total delay improves
-      if (new_drvr_delay + new_fanout_delay
-          < drvr_delay + fanout_delay + fanout_slack) {
-        best_cell = swappable;
-        best_cap = new_fanout_cap;
-        best_area = new_area;
-      }
+    // Use first output for debug display (representative)
+    LibertyPort* first_output_port
+        = swappable->findLibertyPort(output_port_names[0]);
+    float first_new_load_delay
+        = resizer_->gateDelay(first_output_port, output_caps[0], dcalc_ap);
+
+    debugPrint(logger_,
+               RSZ,
+               "size_down",
+               4,
+               " new delay {}->{} gate={} drvr_delta {} + new_delay {} - "
+               "old_delay {} < "
+               "slack {} ({} < {})",
+               network_->pathName(load_pin),
+               network_->pathName(output_pins[0]),
+               swappable->name(),
+               delayAsString(drvr_delta_delay, sta_, 3),
+               delayAsString(first_new_load_delay, sta_, 3),
+               delayAsString(output_delays[0], sta_, 3),
+               delayAsString(slack_margin, sta_, 3),
+               delayAsString(worst_delay_change, sta_, 3),
+               delayAsString(slack_margin, sta_, 3));
+
+    // First case is positive slack and delay change doesn't get worse than
+    // that slack. Second case is negative slack and delay is improved, but
+    // doesn't necessarily fix the slack violation entirely.
+    if ((slack_margin > 0 && worst_delay_change > slack_margin)
+        || (slack_margin < 0 && worst_delay_change > 0)) {
```

Comment:
In the case of flops `slack_margin` will be referring to the input side and `worst_delay_change` to the output side. We should fetch a different slack or ignore sequentials in the move